### PR TITLE
Add support for creating new sections

### PR DESCRIPTION
**Pull request by kentnek**
Hi! I really like this theme and in the process of integerating it into my website, I'd like to contribute a few changes, which will likely come in several PRs.

This first PR makes it easier to add a new section and its corresponding menu, which is what I did for my site.

Please feel free to comment on the changes. Thanks!

### CODE REVIEW

**Comment by jakewies** [comment]
Hey @kentnek ! I'm glad you're enjoying it. This change looks great! I want to test it out locally in a bit before merging, but looks great to me.

File: layouts/partials/nav.html [hunk 1]
```diff
@@ -1,10 +1,10 @@
 <nav class="nav" role="navigation">
   <ul class="nav__list">
-    {{ $currentPage := . }} 
+    {{ $currentPage := . }}
     {{ range .Site.Menus.main }}
       <li>
-        <a {{if or ($currentPage.IsMenuCurrent "main" .) (and (eq .Name "blog") (eq $currentPage.Section "blog")) }} class="active"
-          {{end}} href="{{ .URL }}">{{ .Name }}</a>
+        <a {{ if or ($currentPage.IsMenuCurrent "main" .) (eq $currentPage.Section .Identifier) }} class="active"
+        {{end}} href="{{ .URL }}">{{ .Name }}</a>
```
Comment (jakewies) [inline layouts/partials/nav.html:7]: Sorry I'm terrible with `go` syntax. What's the change here?

**Comment by jakewies** [comment]
Everything looks good! Ready to merge, just want to get some insight into your change at `nav.html` partial so I can understand what's happening now.

File: layouts/partials/nav.html (hunk 1 above)
Comment (kentnek) [reply layouts/partials/nav.html:7]: I was using the solution listed here: https://discourse.gohugo.io/t/highlight-current-menu-item-config-toml/22723/7

Given this menu config:
```toml
[[menu.main]]
  identifier = "about"
  name = "about"
  title = "About"
  url = "/"

[[menu.main]]
  identifier = "blog"
  name = "blog"
  title = "Blog"
  url = "/blog"

```
Basically, the first conditional highlights the homepage (`url = "/"`), and the second conditional highlights whatever menu whose identifier is the same as the page's [section](https://gohugo.io/content-management/sections/#section-page-variables-and-methods) (the first path element in the directory, for example, `/blog/funny/mypost/ => blog`.)

**Comment by kentnek** [comment]
Hi Jake, sorry I went to bed after raising the PR 😂 I've explained the gotemplate line above.

File: layouts/partials/nav.html (hunk 1 above)
Comment (jakewies) [reply layouts/partials/nav.html:7]: Awesome stuff, thanks for the explanation!

**Comment by jakewies** [comment]
Thanks for your contribution @kentnek !

**Comment by jakewies** [comment]
@all-contributors please add @kentnek for code

**Pull request merged**